File: Demo/GtSdkDemo-Manual/GtSdkDemo/Controller/Advance/MessageViewController.m
//
//  MessageViewController.m
//  GtSdkDemo
//
//  Created by ak on 2020/03/20.
//  Copyright © 2019 Gexin Interactive (Beijing) Network Technology Co.,LTD. All rights reserved.
//

#import "MessageViewController.h"

@interface MessageViewController ()
@property (nonatomic, weak) IBOutlet UITextField *messageTextField;
@property (nonatomic, weak) IBOutlet UIButton *ensureSendBtn;
@end

@implementation MessageViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    self.title = @"发送上行消息";
    self.ensureSendBtn.layer.cornerRadius = 6;
    self.messageTextField.attributedPlaceholder = [[NSAttributedString alloc] initWithString:self.messageTextField.placeholder attributes:@{NSForegroundColorAttributeName:[UIColor lightGrayColor]}];
}

- (IBAction)ensureSendMessage:(id)sender {
    NSString *content = _messageTextField.text;
    NSString *msgId = [GeTuiSdk sendMessage:[content dataUsingEncoding:NSUTF8StringEncoding]];
    NSLog(@"%@", msgId);
}

@end

File: Demo/GtSdkDemo-swift/GtSdkDemo-swift/Controller/Advance/MessageViewController.swift
//
//  MessageViewController.swift
//  GtSdkDemo-swift
//
//  Created by ak on 2020/03/20.
//  Copyright © 2020 Gexin Interactive (Beijing) Network Technology Co.,LTD. All rights reserved.
//

import UIKit
import GTSDK

class MessageViewController: UIViewController {
  
  @IBOutlet weak var messageTextField: UITextField!
  @IBOutlet weak var ensureSendMessageBtn: UIButton!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    title = "发送上行消息"
    ensureSendMessageBtn.layer.cornerRadius = 6
    messageTextField.attributedPlaceholder = NSAttributedString(string: messageTextField.placeholder ?? "", attributes: [.foregroundColor: UIColor.lightGray])
  }
  
  @IBAction func ensureSendClick(_ sender: Any) {
    guard let text = messageTextField.text, let data = text.data(using: .utf8) else { return }
    let msgId = try? GeTuiSdk.sendMessage(data)
    NSLog(msgId ?? "")
  }
  
}
